clang/lib/CIR/CodeGen/CIRGenStmt.cpp

```diff
@@ -57,6 +57,7 @@ mlir::LogicalResult CIRGenFunction::emitStmt(const Stmt *s,
 
   switch (s->getStmtClass()) {
   case Stmt::BreakStmtClass:
+  case Stmt::NullStmtClass:
```

Comment:
Classic codegen ALSO puts `NoStmtClass` here, might be worth doing 'while we are here'. In general, only invalid statements should get it, but is there for 'completeness' I presume?  (since there is no default).

Reply:
I'm about 99.9% sure that there is no code that can result in it, barring some error in the compiler.  It is the 'class' of `Stmt`, which isn't `virtual`, but is effectively abstract (other than by language rule).

Reply:
I just looked at classic codegen. It puts NoStmtClass (along with CXXCatchStmtClass, SEHExceptStmtClass, SEHFinallyStmtClass, and MSDependentExistsStmtClass) in a different group just above this that leads to `llvm_unreachable("invalid statement class to emit generically");` rather than `llvm_unreachable("should have emitted these statements as simple");` as we have here. I can still do that here if you want, but it's currently in the group below that goes to an NYI error.

Reply:
Ah, i missed that it wasn't just the same unreachable.  No real reason to do it now, but as it was a 'move this line 10 lines' sorta thing, i thought it was a decent 'while we are here'.

Reply:
I'm fine with that. Trying to avoid doing multiple things, but this is all basically doing nothing in the correct place, so it sort of makes sense.